test/distributed/elastic/rendezvous/c10d_rendezvous_backend_test.py

```diff
@@ -50,43 +78,76 @@ def setUp(self) -> None:
             read_timeout="10",
         )
 
+        # Parameters for filestore testing.
+        self._params_filestore = RendezvousParameters(
+            backend="dummy_backend",
+            endpoint="",
+            run_id="dummy_run_id",
+            min_nodes=1,
+            max_nodes=1,
+            store_type="fIlEsToRe",
+        )
+
         self._expected_endpoint_host = "localhost"
         self._expected_endpoint_port = 29300
         self._expected_store_type = TCPStore
         self._expected_read_timeout = timedelta(seconds=10)
 
-    def test_create_backend_returns_backend(self) -> None:
+    def run_test_with_store(self, store_type: str, test_to_run: Callable):
+        """
+        Use this function to specify the store type to use in a test. If
+        not used, the test will default to TCPStore.
+        """
+        if store_type == "filestore":
+            self._params = self._params_filestore
+            self._expected_store_type = FileStore
+            # in case there is a test which defaults to tcp store, we set the
+            # default tcp parameters
+            self._expected_endpoint_port = 29400
+            self._expected_read_timeout = timedelta(seconds=60)
+
+        test_to_run()
+
+    def create_backend_returns_backend(self) -> None:
         backend, store = create_backend(self._params)
 
         self.assertEqual(backend.name, "c10d")
 
         self.assertIsInstance(store, self._expected_store_type)
 
-        tcp_store = cast(TCPStore, store)
-
-        self.assertEqual(tcp_store.host, self._expected_endpoint_host)  # type: ignore[attr-defined]
-        self.assertEqual(tcp_store.port, self._expected_endpoint_port)  # type: ignore[attr-defined]
-        self.assertEqual(tcp_store.timeout, self._expected_read_timeout)  # type: ignore[attr-defined]
+        tcp_store = cast(self._expected_store_type, store)
+        if (self._expected_store_type == TCPStore):
+            self.assertEqual(tcp_store.host, self._expected_endpoint_host)  # type: ignore[attr-defined]
+            self.assertEqual(tcp_store.port, self._expected_endpoint_port)  # type: ignore[attr-defined]
+            self.assertEqual(tcp_store.timeout, self._expected_read_timeout)  # type: ignore[attr-defined]
```

Comment:
So for that, FileStore does not take in timeout as an option (which makes sense, since the store is based on a local file). In testing, it seems that the Store timeout property ends up defaulting to 300 seconds instead of the 60 seconds that it says it should be in documentation. For that reason I had left it out of FileStore tests.

Reply:
Yes, `FileStore` does not have a timeout argument in its constructor, but all classes derived from `Store` have the timeout property. The timeout is still used in FileStore for all operations when accessing the file, e.g. (`get`, `est`, `add`, etc.), here's a pointer if curious https://github.com/pytorch/pytorch/blob/master/torch/lib/c10d/FileStore.cpp#L154-L158. Also wondering where in the docs it says 60 seconds, probably an error, can you link to me?

Reply:
Ah ok. Here's the documentation I was referring to: https://github.com/pytorch/pytorch/blob/master/torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py#L182
I believe that it said 60 there because there was some logic in _create_tcp_store that defaults it to 60 if not specified:
https://github.com/pytorch/pytorch/blob/master/torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py#L137

Reply:
I see, thought you meant official pytorch docs. You can also set timeout to be 60 in your internal `_create_file_store` so it matches.